pkg/api/resource_helpers.go

```diff
@@ -80,12 +81,44 @@ func IsPodReadyConditionTrue(status PodStatus) bool {
 // Extracts the pod ready condition from the given status and returns that.
 // Returns nil if the condition is not present.
 func GetPodReadyCondition(status PodStatus) *PodCondition {
+	_, condition := GetPodCondition(&status, PodReady)
+	return condition
+}
+
+func GetPodCondition(status *PodStatus, conditionType PodConditionType) (index int, condition *PodCondition) {
```

Comment:
is there some reason why you used variable names in the return types? you don't seem to be using the feature, i.e. (int, *PodCondition) would be equivalent (actually I'm surprised what you have works)